app/screens/participants_list/participant_row/index.tsx

```diff
@@ -0,0 +1,142 @@
+import {ImageSource} from 'react-native-vector-icons/Icon';
+
+import {showModal} from '@actions/navigation';
+import CompassIcon from '@components/compass_icon';
+import ProfilePicture from '@components/profile_picture';
+import type {Theme} from '@mm-redux/types/preferences';
+import {UserProfile} from '@mm-redux/types/users';
+import {displayUsername} from '@mm-redux/utils/user_utils';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+interface ParticipantRowProps {
+        teammateNameDisplay: string,
+        theme: Theme,
+        user: UserProfile,
+    }
+
+export default class ParticipantRow extends React.PureComponent<ParticipantRowProps> {
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+    closeButton: ImageSource
+
+    goToUserProfile = async () => {
+        const {user, theme} = this.props;
+        const {formatMessage} = this.context.intl;
+        const screen = 'UserProfile';
+        const title = formatMessage({id: 'mobile.routes.user_profile', defaultMessage: 'Profile'});
+        const passProps = {
+            userId: user.id,
+        };
+
+        if (!this.closeButton) {
+            this.closeButton = await CompassIcon.getImageSource('close', 24, theme.sidebarHeaderTextColor);
+        }
+
+        const options = {
+            topBar: {
+                leftButtons: [{
+                    id: 'close-settings',
+                    icon: this.closeButton,
+                }],
+            },
+        };
+
+        showModal(screen, title, passProps, options);
+    };
+
+    render() {
+        const {
+            teammateNameDisplay,
+            user,
+            theme,
+        } = this.props;
+
+        if (!user.id) {
+            return null;
+        }
+
+        const {id, username} = user;
+        const usernameDisplay = '@' + username;
+
+        const style = getStyleSheet(theme);
+
+        return (
+            <TouchableOpacity
+                key={user.id}
+                onPress={preventDoubleTap(this.goToUserProfile)}
+            >
+                <View style={style.container}>
+                    <View style={style.profileContainer}>
+                        <View style={style.profile}>
+                            <ProfilePicture
+                                userId={id}
+                                showStatus={false}
+                                size={24}
+                            />
+                        </View>
+                    </View>
+                    <Text
+                        style={style.textContainer}
+                        ellipsizeMode='tail'
+                        numberOfLines={1}
+                    >
+                        <Text style={style.displayName}>
+                            {displayUsername(user, teammateNameDisplay)}
+                        </Text>
+                        <Text>{'  '}</Text>
```

Comment:
do we need this? maybe just add a marginLeft to `style.username`? One other thing is, what happens if `teammateNameDisplay` is set to username, we would be displaying the username twice, right?

Reply:
Hehe, I had issues styling this with marginLift so I took "inspiration" from the reaction_list component which does the same thing.

Yep, if `teammateDisplay` is set to username then it will show username twice. This component is inspired by the reaction_row component, therefore that suffers from this problem too.
Perhaps relying on existing components for inspiration for implementation isn't right?

Reply:
If that is the case, I would create a component (just one) and then have both this list and the reaction list using the same component instead of providing inspiration, as in the end, they both should be the same, yes?

Reply:
They are similar, yes. I didn't want to get into refactoring the existing component. 
The order of the reaction row elements are different (there is a proposed redesign which will make them diverge further so having a single component might be messy. See Matt's comment above).

Reply:
>do we need this? maybe just add a marginLeft to style.username?

This is a text component inside a text component, so [flexbox layout no longer works](https://reactnative.dev/docs/text#containers). Wrapping a child Text component in a View component breaks the single line directive in the parent Text component that forces everything on a single line. This seems like a simple fix around this problem.


>One other thing is, what happens if teammateNameDisplay is set to username, we would be displaying the username twice, right?

I've hardcoded it to display the full name as in the figma design

Reply:
It is not about hardcoding it, is cause the user may not set their first & lastname, so the function here will return the username as default anyway, so hardcoding does not really solve the problem.

Reply:
Are you sure about the layout? For some reason that seems odd to me

Reply:
In any case think of just using that or adding an additional space at the first of the next Text when applicable

Reply:
>It is not about hardcoding it, is cause the user may not set their first & lastname, so the function here will return the username as default anyway, so hardcoding does not really solve the problem.

I think this might be something for the UI team to decide? 

@matthewbirtch - What do we fallback to if the user has not set a first and last name? The code defaults to displaying the username if the first and last names have not been set. This looks odd because then the username is displayed twice next to each other.

Reply:
@matthewbirtch thoughts on what we want the screen to show when the user doesn't have a first and last name set? Currently it defaults to the username, in which case the screen shows the username twice.

Reply:
> @matthewbirtch thoughts on what we want the screen to show when the user doesn't have a first and last name set? Currently it defaults to the username, in which case the screen shows the username twice.

We definitely don't want to show the username twice. If the First name/Last name is not set, we should only show the username, but style it the same as the display name